### PR TITLE
Fixes #963: Comparison Style as global settings (aka Yoda settings)

### LINKED ISSUE #963: Global Yoda/Regular switch
Is it possible to set a global setting for Yoda vs Regular in the main settings for the plugin (PhpStorm settings > Other settings) that will change all of the inspections options together?

I've been using PHP Inspections EA for over month now, and really like it. The only small annoyance I had was that applying automatic fixes were done with Yoda-style conditions, but I'd do the fix and then swap the condition to regular style.

It wasn't until today (after doing that process 100+ times) that I discovered that those inspections have an option for changing from "Yoda fix style" to "Regular fix style". I'd rather not have to scan through every inspection to change them all to Regular one-by-one.

If there had been a global setting I'd have found it on the first day of using the plugin.

### PR DESCRIPTION
**Pull request by rentalhost**
Due to amount of related files (that have implemented an exclusive Yoda style) I have did a lot of things on the same PR. But I believe that it could be considered as positive here, else will be need do a lot of PR to make it happen.

- [x] **Status**: finished.
  - [x] **Note:** rebased to last master [commit](https://github.com/kalessil/phpinspectionsea/commit/6887a3e8e31a3558a27a648963e67d0cb6726aed) at 2018-07-08;
- [x] **ComparisonStyle** _new_ class:
  - [x] **Added:** helper to controls and reports current Comparison Style;
- [x] **EAApplicationConfiguration** class:
  - [x] **Added:** moved Comparison Style setting to here grouped into _Comparison code style_;
  - [x] **Improvement:** group _Anonymous data collect_ options;
- [x] **EASettings** class:
  - [x] **Added**: persists the new Comparison Style setting (default to _Regular_);
  - [x] **Fixed:** `if()` using `versionOldest` instead of `version`;
- [x] **UnSafeIsSetOverArrayInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: modified some tests from Yoda style to Regular style;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **SubStrUsedAsStrPosInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: modified some tests from Yoda style to Regular style;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **IsNullFunctionUsageInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **IsEmptyFunctionUsageInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: modified some tests from Yoda style to Regular style;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **InArrayMissUseInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: modified some tests from Yoda style to Regular style;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **ComparisonOperandsOrderInspector** class:
  - [x] **Improvement:** use Comparison Style from global setting;
  - [x] **Tests**: passed;
  - [x] **Manual tests:** passed;
- [x] **OptionsComponent** class:
  - [x] **Added:** new `addHyperlink()` method to create an hyperlinked label to configuration page;
  - [x] **Added:** new `addPanel(label)` to create a new panel with a titled separator (_group_);

### CODE REVIEW

**Comment by kalessil** [comment]
Great work, thank you @rentalhost ! Taking care about the PR today.

**Comment by rentalhost** [comment]
There are some other settings that should be reworked like that. For instance, there some cases here the autofix creates an `in_array()`, and it doesn't respects the `\in_array()` option (which the slash). So I think that it could be improved to this way, if you could process this PR until next weekend, I will make it possible.

**Comment by kalessil** [comment]
Sounds awesome! On my end this evening should be enough to check and merge it.

File: src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/IsEmptyFunctionUsageInspector.java [hunk 1]
```diff
@@ -120,27 +118,14 @@ private boolean isNullableCoreType(@NotNull Set<String> resolvedTypesSet) {
...
+                       resolvedTypesSet.contains(Types.strResource);
             }
         };
     }
 
-    public JComponent createOptionsPanel() {
-        return OptionsComponent.create((component) -> {
-            component.addCheckbox("Report empty() usage", REPORT_EMPTY_USAGE, (isSelected) -> REPORT_EMPTY_USAGE = isSelected);
-            component.addCheckbox("Suggest to use count()-comparison", SUGGEST_TO_USE_COUNT_CHECK, (isSelected) -> SUGGEST_TO_USE_COUNT_CHECK = isSelected);
-            component.addCheckbox("Suggest to use null-comparison", SUGGEST_TO_USE_NULL_COMPARISON, (isSelected) -> SUGGEST_TO_USE_NULL_COMPARISON = isSelected);
```
Comment (kalessil) [inline src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/IsEmptyFunctionUsageInspector.java]: @rentalhost : are those settings gone completely, or you've relocated them somewhere?

File: src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeStyle/ComparisonOperandsOrderInspector.java [hunk 2]
```diff
@@ -71,10 +72,9 @@ public void visitPhpBinaryExpression(@NotNull BinaryExpression expression) {
     }
 
     public JComponent createOptionsPanel() {
-        return OptionsComponent.create((component) -> component.delegateRadioCreation((radioComponent) -> {
-            radioComponent.addOption("Prefer regular style", PREFER_REGULAR_STYLE, (isSelected) -> PREFER_REGULAR_STYLE = isSelected);
-            radioComponent.addOption("Prefer yoda style", PREFER_YODA_STYLE, (isSelected) -> PREFER_YODA_STYLE = isSelected);
-        }));
+        return OptionsComponent.create((component) -> {
+            component.addHyperlink("Setup Yoda or Regular style...", EAApplicationConfiguration.class);
+        });
```
Comment (kalessil) [inline src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/codeStyle/ComparisonOperandsOrderInspector.java]: Awesome solution, thanks!

File: src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/IsEmptyFunctionUsageInspector.java (hunk 1 above)
Comment (rentalhost) [reply src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/apiUsage/IsEmptyFunctionUsageInspector.java]: Done. I have checked other inspections and seems that it was the unique issue (uff!).

**Comment by kalessil** [comment]
Thank you @rentalhost !

**Pull request merged**